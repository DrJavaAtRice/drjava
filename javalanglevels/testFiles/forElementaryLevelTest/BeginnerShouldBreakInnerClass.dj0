/*!DrJava Beginner Level*/

class BeginnerShouldBreakInnerClass extends Object {
  class InnerClass extends Object {
    int j;
    int k;
  }
}