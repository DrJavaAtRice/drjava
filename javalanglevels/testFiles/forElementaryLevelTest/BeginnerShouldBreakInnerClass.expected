// Language Level Converter line number map: dj*->java. Entries: 8
//     1->3         2->4         3->5         4->6         5->7         6->8         7->9         8->57   
/*!DrJava Beginner Level*/

class BeginnerShouldBreakInnerClass extends Object {
  class InnerClass extends Object {
    private final int j;
    private final int k;
  
    /** This method is automatically generated by the Language Level Converter. */
    public InnerClass(int j, int k) {
      super();
      this.j = j;
      this.k = k;
    }

    /** This method is automatically generated by the Language Level Converter. */
    public int j() {
      return j;
    }

    /** This method is automatically generated by the Language Level Converter. */
    public int k() {
      return k;
    }

    /** This method is automatically generated by the Language Level Converter. */
    public java.lang.String toString() {
      return getClass().getName() + "(" + 
          j() + ", " + 
          k() + 
          ")";
    }

    /** This method is automatically generated by the Language Level Converter. */
    public boolean equals(java.lang.Object o) {
      if (this == o) {
        return true;
      }
      else if ((o == null) || (! o.getClass().equals(getClass()))) {
        return false;
      }
      else {
        InnerClass cast = ((InnerClass) o);
        return (j() == cast.j()) && 
            (k() == cast.k());
      }
    }

    /** This method is automatically generated by the Language Level Converter. */
    public int hashCode() {
      return getClass().hashCode() ^ 
          j() ^ 
          k();
    }
  }

  /** This method is automatically generated by the Language Level Converter. */
  public BeginnerShouldBreakInnerClass() {
    super();
  }

  /** This method is automatically generated by the Language Level Converter. */
  public java.lang.String toString() {
    return getClass().getName() + "(" + 
        ")";
  }

  /** This method is automatically generated by the Language Level Converter. */
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    else if ((o == null) || (! o.getClass().equals(getClass()))) {
      return false;
    }
    else {
      BeginnerShouldBreakInnerClass cast = ((BeginnerShouldBreakInnerClass) o);
      return true;
    }
  }

  /** This method is automatically generated by the Language Level Converter. */
  public int hashCode() {
    return getClass().hashCode();
  }
}
